lib/pfconfig/cached_hash.pm

```diff
@@ -177,6 +177,46 @@ sub EXISTS {
     return $self->_get_from_socket( $self->{_namespace}, "key_exists", ( search => $key ) )->{result};
 }
 
+=head2 values
+
+Added method that can be called on the underlying object of the tied hash
+Will return all the values of the hash. Mostly for internal use
+Call it using tied(%hash)->values
+
+=cut
+
+sub values {
```

Comment:
You're right but the thing is using FETCH uses the subcache. This means that next calls to values will be done without poking pfconfig. Also, the subcache may have already been built so it'll be faster this way IMO. I could rewrite it so it pokes pfconfig and builds the subcache with the result but i don't think it'll do a big difference for the code it adds